src/diffpy/labpdfproc/tools.py

```diff
@@ -1,7 +1,17 @@
+import sys
+
 WAVELENGTHS = {"Mo": 0.71, "Ag": 0.59, "Cu": 1.54}
 
 
 def set_wavelength(args):
+    if args.wavelength and not isinstance(args.wavelength, float):
+        sys.exit("Please enter the wavelength as a float value.")
+    if not args.wavelength and args.anode_type and not (args.anode_type in WAVELENGTHS):
+        sys.exit(
+            "Please either specify a wavelength as a float value "
+            "or specify anode_type as one of 'Mo', 'Ag', or 'Cu'."
```

Comment:
use an f-string and unpack WAVELENGTHS so when we update wavelengths the warning message updates automatically.

Also specify it as an error and tell teh user what to do:  "please rerun the program specifying...."  The best may be to throw a `ValueError` Exception rather than sys.exit which is not good practice.

Reply:
Edited. I also added tests for invalid anode types and non-positive wavelengths. I find these two lines

WAVELENGTHS = {"Mo": 0.71, "Ag": 0.59, "Cu": 1.54}
known_sources = [key for key in WAVELENGTHS.keys()]

appear in both labpdfprocapp.py and tools.py, but we need known_sources in the help message of anode type. Maybe we should just move them to tools.py?